src/components/program/ProgramContentMaterialBlock.tsx

```diff
@@ -80,86 +78,91 @@ const ProgramContentMaterialBlock: React.VFC<{
-              onClick={async () => {
-                await insertMaterialAuditLog({
-                  variables: {
-                    data: { member_id: currentMemberId, target: material.id, action: 'download' },
-                  },
-                })
-                  .then(async res => {
-                    if (res.data?.insert_material_audit_log_one?.id) {
-                      if (material.data.url) {
-                        window.open(material.data.url)
-                        return
-                      }
-                      setIsDownloading(prev => ({ ...prev, [material.id]: true }))
-                      const materialLink = await getFileDownloadableLink(
-                        `materials/${appId}/${programContentId}_${material.data.name}`,
-                        authToken,
+          .map(
+            material =>
+              material.programContentId === programContentId && (
+                <StyledMaterial
+                  key={material.id}
+                  className="mb-3"
+                  onClick={async () => {
+                    await insertMaterialAuditLog({
+                      variables: {
+                        data: { member_id: currentMemberId, target: material.id, action: 'download' },
+                      },
+                    })
+                      .then(async res => {
+                        if (res.data?.insert_material_audit_log_one?.id) {
+                          if (material.data.url) {
+                            window.open(material.data.url)
+                            return
+                          }
+                          setIsDownloading(prev => ({ ...prev, [material.id]: true }))
+                          const materialLink = await getFileDownloadableLink(
+                            `materials/${appId}/${programContentId}_${material.data.name}`,
+                            authToken,
+                          )
+                          const materialRequest = new Request(materialLink)
+                          try {
+                            const response = await fetch(materialRequest)
+                            response.url &&
+                              downloadFile(material.data.name, {
+                                url: response.url,
+                                onDownloadProgress: ({ loaded, total }) => {
+                                  setDownloadProgress(prev => ({
+                                    ...prev,
+                                    [material.id]: Math.floor((loaded / total) * 100),
+                                  }))
+                                },
+                              }).then(() => {
+                                setIsDownloading(prev => ({ ...prev, [material.id]: false }))
+                              })
+                          } catch (error) {
+                            process.env.NODE_ENV === 'development' && console.error(error)
+                          }
```

Comment:
mixing async/await with then/catch here , using just async/await could make your code cleaner and easier

Reply:
Okay, I have adjusted it, and also changed it to handle function to help better reading.